### DISCUSSION TITLE
Can AG vision read dicom images?

### DISCUSSION THREAD

**Discussion by talhaanwarch** [autogluon/autogluon#1233] [Q&A] [answered]
I want to know can autogluon read dicom images and what image extension are supported?

**Comment by Innixma** [comment, marked as answer]
While I'm not the most familiar with dicom images, if they are 3D then I don't think AG supports them by default. If they are 2D, then it is just a matter of converting the image format to one that is supported, such as png or jpg. If you'd like to request support for them, feel free to open a new github issue with more information about the format and perhaps a reference to a dataset which leverages dicom images (and ideally one that has a known baseline solution such as a kaggle competition).

**Reply by Innixma** [reply]
@zhreshold feel free to expand on this if I missed anything

**Reply by talhaanwarch** [reply]
yes, it's 2D images, and there is a[ recent Kaggle competition](https://www.kaggle.com/c/rsna-miccai-brain-tumor-radiogenomic-classification) for this which has just started. Recently, I participated in CT-scan image classification using AG and the results are really amazing. Thats why I want to give here a try

**Reply by Innixma** [reply]
Given its a 2D image, we should have no problems. If we don't natively support the file format, you could try implementing a file type converter / image loader. If it is simple enough, it might also be worthwhile contributing it to AG so we can support the dicom format natively in future.

Great to hear that AG did well for you in CT-scan classification!

**Reply by talhaanwarch** [reply]
yea, i can convert dicom to png/jpg, but I have to save data to disk first and then load it again via AG. Can you link to a reference where data loading takes place

**Reply by Innixma** [reply]
I'm unsure regarding images, it might actually be in GluonCV. @zhreshold

**Reply by zhreshold** [reply]
@talhaanwarch the question can be simplified to "can opencv or PIL read DICOM images?". 
After doing some research on the internet, I found there seems to be a dedicated package `dicom` for reading such format in python, neither opencv or PIL can read them directly, so dicom images are not supported by ag right now. 
Is there any blocker for converting dicom images to normal JPEG/PNG images? @talhaanwarch

**Reply by Innixma** [reply]
@talhaanwarch 

I think that it might be the case that dicom is niche enough to not warrant dedicated support in AG. As Zhi said, if you can convert them manually to jpeg/png, that might be best. If you think its important to have dedicated support, feel free to open a GitHub issue with the feature request and if we get enough community interest we could look into adding it.

**Reply by talhaanwarch** [reply]
I can convert them manually, but for this first, I have to save them to disk, I am just trying to avoid this. 
What I am trying to say is 
I am just looking for a custom function that passes to `fit` function along with data frame (as data frame is already an argument). this custom function overrides the default image reading function.
It can be simple as 
```
def read_image(dataframe,index):
  img=cv2.imread(dataframe.image[index])
  return img
```
and user can define it no matter it is Dicom image and other non supported images by AG. It will help to read all images not supported by AG but supported by other python packages.
You can consider it as a requests for feature version
